src/gt4py/next/program_processors/codegens/gtfn/gtfn_ir_to_gtfn_im_ir.py

```diff
@@ -264,25 +264,23 @@ def visit_FunCall(self, node: gtfn_ir.FunCall, **kwargs: Any) -> gtfn_ir_common.
             self.imp_list_ir.append(InitStmt(lhs=gtfn_ir_common.Sym(id=f"{lam_idx}"), rhs=node))
             return gtfn_ir_common.SymRef(id=f"{lam_idx}")
         if isinstance(node.fun, gtfn_ir.Lambda):
+            localized_symbols = {**kwargs["localized_symbols"]}  # create a new scope
+            new_kwargs = {**kwargs, "localized_symbols": localized_symbols}
```

Comment:
I would move the definition of `new_kwargs` to the line before is used, to make it simpler to read.